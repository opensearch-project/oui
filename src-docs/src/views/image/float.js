/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import { OuiImage, OuiText } from '../../../../src/components';
import { faker } from '@faker-js/faker';

export default () => (
  <OuiText>
    <OuiImage
      size="l"
      float="right"
      margin="l"
      hasShadow
      caption="Random nature image"
      allowFullScreen
      alt="Random nature image"
      src="https://picsum.photos/800/500"
    />
    <p>{faker.helpers.fake('{{lorem.paragraphs}}')}</p>
    <p>{faker.helpers.fake('{{lorem.paragraphs}}')}</p>
    <p>{faker.helpers.fake('{{lorem.paragraphs}}')}</p>
    <OuiImage
      size="l"
      float="left"
      margin="l"
      hasShadow
      allowFullScreen
      caption="Another random image"
      alt="Random nature image"
      src="https://picsum.photos/300/300"
    />
    <p>{faker.helpers.fake('{{lorem.paragraphs}}')}</p>
    <p>{faker.helpers.fake('{{lorem.paragraphs}}')}</p>
  </OuiText>
);
