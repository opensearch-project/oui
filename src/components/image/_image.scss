/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/**
 * 1. Fix for IE where the image correctly resizes in width but doesn't collapse its height
      (https://github.com/philipwalton/flexbugs/issues/75#issuecomment-134702421)
 */

// Main <figure> that wraps images.
.ouiImage {
  display: inline-block;
  max-width: 100%;
  position: relative;
  min-height: 1px; /* 1 */
  line-height: 0; // Fixes cropping when image is resized by forcing its height to be determined by the image not line-height
  flex-shrink: 0; // Don't ever let this shrink in height if direct descendent of flex

  // Required for common usage of nesting within OuiText
  .ouiImage__img {
    margin-bottom: 0;
  }

  &.ouiImage--hasShadow {
    .ouiImage__img {
      @include ouiBottomShadowMedium;
    }
  }

  .ouiImage__button {
    position: relative;
    cursor: pointer;

    // transition the shadow
    transition: all $ouiAnimSpeedFast $ouiAnimSlightResistance;

    &:focus  {
      outline: 2px solid $ouiFocusRingColor;
    }

    &:hover .ouiImage__icon {
      visibility: visible;
      fill-opacity: 1;
    }

    &--fullWidth {
      width: 100%;
    }
  }

  &.ouiImage--allowFullScreen {
    &:hover .ouiImage__caption {
      text-decoration: underline;
    }

    &:not(.ouiImage--hasShadow) .ouiImage__button:hover,
    &:not(.ouiImage--hasShadow) .ouiImage__button:focus {
      @include ouiBottomShadowMedium;
    }

    &.ouiImage--hasShadow .ouiImage__button:hover,
    &.ouiImage--hasShadow .ouiImage__button:focus {
      @include ouiBottomShadow;
    }
  }

  // These sizes are mostly suggestions. Don't look too hard for meaning in their values.
  // Size is applied to the image, rather than the figure to work better with floats
  &.ouiImage--small .ouiImage__img {
    width: convertToRem(120px);
  }

  &.ouiImage--medium .ouiImage__img {
    width: convertToRem(200px);
  }

  &.ouiImage--large .ouiImage__img {
    width: convertToRem(360px);
  }

  &.ouiImage--xlarge .ouiImage__img {
    width: convertToRem(600px);
  }

  &.ouiImage--fullWidth {
    width: 100%;
  }

  &.ouiImage--original {
    .ouiImage__img {
      width: auto;
      max-width: 100%;
    }
  }

  &.ouiImage--floatLeft {
    float: left;

    &[class*='ouiImage--margin']  {
      margin-left: 0;
      margin-top: 0;
    }
  }

  &.ouiImage--floatRight {
    float: right;

    &[class*='ouiImage--margin']  {
      margin-right: 0;
      margin-top: 0;
    }
  }

  &.ouiImage--marginSmall {
    margin: $ouiSizeS;
  }

  &.ouiImage--marginMedium {
    margin: $ouiSize;
  }

  &.ouiImage--marginLarge {
    margin: $ouiSizeL;
  }

  &.ouiImage--marginXlarge {
    margin: $ouiSizeXL;
  }
}

// The image itself is full width within the container.
.ouiImage__img {
  width: 100%;
  vertical-align: middle;
}

.ouiImage__caption {
  @include ouiFontSizeS;

  margin-top: $ouiSizeXS;
  text-align: center;
}

.ouiImage__icon {
  visibility: hidden;
  fill-opacity: 0;
  position: absolute;
  right: $ouiSize;
  top: $ouiSize;
  transition: fill-opacity $ouiAnimSpeedSlow $ouiAnimSlightResistance;
  cursor: pointer;
}

// The FullScreen image that optionally pops up on click.
.ouiImage-isFullScreen {
  position: relative;
  max-height: 80vh;
  max-width: 80vw;
  animation: ouiImageFullScreen $ouiAnimSpeedExtraSlow $ouiAnimSlightBounce;

  &:hover {
    .ouiImage__button {
      @include ouiBottomShadow;
    }

    .ouiImage__caption {
      text-decoration: underline;
    }
  }

  &__img {
    max-height: 80vh;
    max-width: 80vw;
    vertical-align: middle;
    cursor: pointer;
    transition: all $ouiAnimSpeedFast $ouiAnimSlightResistance;
  }
}

.ouiImage-isFullScreenCloseIcon {
  position: absolute;
  right: $ouiSize;
  top: $ouiSize;
  pointer-events: none;
}

@keyframes ouiImageFullScreen {
  0% {
    opacity: 0;
    transform: translateY($ouiSizeXL * 2);
  }

  100% {
    opacity: 1;
    transform: translateY(0);
  }
}

@include ouiBreakpoint('xs', 's', 'm') {
  .ouiImage {
    &.ouiImage--floatLeft,
    &.ouiImage--floatRight {
      float: none;

      // Return back to whatever margin settings were set without the float
      &[class*='ouiImage--margin']  {
        margin-top: inherit;
        margin-right: inherit;
        margin-bottom: inherit;
        margin-left: inherit;
      }
    }
  }
}
