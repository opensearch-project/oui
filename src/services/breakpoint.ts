/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { keysOf } from '../components/common';

export type OuiBreakpointSize = 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'xxxl';

export type OuiBreakpoints = {
  /**
   * Set the minimum window width at which to start to the breakpoint
   */
  [key in OuiBreakpointSize]: number;
};

export const BREAKPOINTS: OuiBreakpoints = {
  xxxl: 1920,
  xxl: 1680,
  xl: 1200,
  l: 992,
  m: 768,
  s: 575,
  xs: 0,
};

export const BREAKPOINT_KEYS = keysOf(BREAKPOINTS);

/**
 * Given the current `width` and an object of `OuiBreakpoints`,
 * this function returns the string that is the name of the breakpoint key
 * that is less than or equal to the width
 *
 * @param {number} width Can either be the full window width or any width
 * @param {OuiBreakpoints} breakpoints An object with keys for sizing and values for minimum width
 * @param {OuiBreakpointSize[] | undefined} allowedBreakpoints An array of allowed breakpoints
 * @returns {string | undefined} Name of the breakpoint key or `undefined` if a key doesn't exist
 */
export function getBreakpoint(
  width: number,
  breakpoints: OuiBreakpoints = BREAKPOINTS,
  allowedBreakpoints: OuiBreakpointSize[] = keysOf(BREAKPOINTS)
): OuiBreakpointSize | undefined {
  // Find the breakpoint (key) whose value is <= windowWidth starting with the largest first
  return keysOf(BREAKPOINTS).find(
    (key) => allowedBreakpoints.includes(key) && breakpoints[key] <= width
  );
}

/**
 * Given the current `width` and a max breakpoint key,
 * this function returns true or false if the `width` falls within the max
 * breakpoint or any breakpoints below
 *
 * @param {number} width Can either be the full window width or any width
 * @param {OuiBreakpointSize | number} max The named breakpoint or custom number to check against
 * @param {OuiBreakpoints} breakpoints An object with keys for sizing and values for minimum width
 * @returns {boolean} Will return `false` if it can't find a value for the `max` breakpoint
 */
export function isWithinMaxBreakpoint(
  width: number,
  max: OuiBreakpointSize | number,
  breakpoints: OuiBreakpoints = BREAKPOINTS
): boolean {
  if (typeof max === 'number') {
    return width <= max;
  } else {
    const currentBreakpoint = getBreakpoint(width, breakpoints);
    return currentBreakpoint
      ? breakpoints[currentBreakpoint] <= breakpoints[max]
      : false;
  }
}

/**
 * Given the current `width` and a max breakpoint key,
 * this function returns true or false if the `width` falls within the max
 * breakpoint or any breakpoints below
 *
 * @param {number} width Can either be the full window width or any width
 * @param {OuiBreakpointSize | number} min The named breakpoint or custom number to check against
 * @param {OuiBreakpoints} breakpoints An object with keys for sizing and values for minimum width
 * @returns {boolean} Will return `false` if it can't find a value for the `min` breakpoint
 */
export function isWithinMinBreakpoint(
  width: number,
  min: OuiBreakpointSize | number,
  breakpoints: OuiBreakpoints = BREAKPOINTS
): boolean {
  if (typeof min === 'number') {
    return width >= min;
  } else {
    const currentBreakpoint = getBreakpoint(width, breakpoints);
    return currentBreakpoint
      ? breakpoints[currentBreakpoint] >= breakpoints[min]
      : false;
  }
}

/**
 * Given the current `width` and an array of breakpoint keys,
 * this function returns true or false if the `width` falls within
 * any of the named breakpoints
 *
 * @param {number} width Can either be the full window width or any width
 * @param {OuiBreakpointSize[]} sizes An array of named breakpoints
 * @param {OuiBreakpoints} breakpoints An object with keys for sizing and values for minimum width
 * @returns {boolean} Returns `true` if current breakpoint name is included in `sizes`
 */
export function isWithinBreakpoints(
  width: number,
  sizes: OuiBreakpointSize[],
  breakpoints: OuiBreakpoints = BREAKPOINTS
): boolean {
  const currentBreakpoint = getBreakpoint(width, breakpoints);
  return currentBreakpoint ? sizes.includes(currentBreakpoint) : false;
}

/* OUI -> EUI Aliases */
export type EuiBreakpointSize = OuiBreakpointSize;
export type EuiBreakpoints = OuiBreakpoints;
/* End of Aliases */
