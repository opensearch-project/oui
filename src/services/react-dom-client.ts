/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import * as ReactDOM from 'react-dom';

/**
 * Root interface for React 16 compatibility with React 18's createRoot API
 */
export interface Root {
  render(element: React.ReactElement): void;
  unmount(): void;
}

/**
 * Shim for React 18's createRoot API
 * This provides a compatible API for React 16 using ReactDOM.render
 *
 * @param container The DOM element to render into
 * @returns A root object with render and unmount methods
 */
export function createRoot(container: Element): Root {
  return {
    render(element: React.ReactElement) {
      ReactDOM.render(element, container);
    },
    unmount() {
      ReactDOM.unmountComponentAtNode(container);
    },
  };
}
