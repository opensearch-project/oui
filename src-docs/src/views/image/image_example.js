/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import { renderToHtml } from '../../services';

import { GuideSectionTypes } from '../../components';

import { OuiCode, OuiImage } from '../../../../src/components';
OuiImage.__docgenInfo.props.src.required = true;

import imageConfig from './playground';

import Image from './image';
const imageSource = require('./image?raw');
const imageHtml = renderToHtml(Image);
const imageSnippet = `<OuiImage
  alt={description}
  src={someSrc}
/>
`;

import ImageSizes from './image_size';
const imageSizesSource = require('./image_size?raw');
const imageSizesHtml = renderToHtml(ImageSizes);
const imageSizesSnippet = `<OuiImage
  size="l"
  alt={description}
  src={someSrc}
/>
`;

import ImageZoom from './image_zoom';
const imageZoomSource = require('./image_zoom?raw');
const imageZoomHtml = renderToHtml(ImageZoom);
const imageZoomSnippet = `<OuiImage
  allowFullScreen
  alt={description}
  src={someSrc}
/>
`;

import ImageFloat from './float';
import { OuiCallOut } from '../../../../src/components/call_out';
import { Fragment } from 'react-is';
const imageFloatSource = require('./float?raw');
const imageFloatHtml = renderToHtml(ImageFloat);
const imageFloatSnippet = `<OuiImage
  alt={description}
  src={someSrc}
  float="left"
  margin="l"
/>
`;

export const ImageExample = {
  title: 'Image',
  sections: [
    {
      source: [
        {
          type: GuideSectionTypes.JS,
          code: imageSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: imageHtml,
        },
      ],
      text: (
        <div>
          <p>
            Use <strong>OuiImage</strong> when you need to place a static image
            into a page with an optional caption.
          </p>
        </div>
      ),
      props: { OuiImage },
      demo: <Image />,
      snippet: imageSnippet,
    },
    {
      title: 'Click an image for a full screen version',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: imageZoomSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: imageZoomHtml,
        },
      ],
      text: (
        <p>
          Apply the <OuiCode>allowFullScreen</OuiCode> prop to make the image
          clickable and show a full screen version. Note that the second image
          also passes{' '}
          <OuiCode language="js">fullScreenIconColor=&quot;dark&quot;</OuiCode>{' '}
          to change icon color to better contrast against the light background
          of that image.
        </p>
      ),
      demo: <ImageZoom />,
      snippet: imageZoomSnippet,
    },
    {
      title: 'Images can be sized',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: imageSizesSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: imageSizesHtml,
        },
      ],
      text: (
        <p>
          Images can be sized by passing the <OuiCode>size</OuiCode> prop a
          value of{' '}
          <OuiCode>
            s / m / l / xl / original / fullWidth / number / string
          </OuiCode>
          . This size sets the <strong>maximum</strong> length of the longest
          edge of the image, whether that is height or width, and scales it.
          Only the provided sizing values will also increase the size of a
          smaller image.
        </p>
      ),
      demo: <ImageSizes />,
      snippet: imageSizesSnippet,
    },
    {
      title: 'Float images within text',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: imageFloatSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: imageFloatHtml,
        },
      ],
      text: (
        <Fragment>
          <p>
            When using <OuiCode>OuiImage</OuiCode> within{' '}
            <OuiCode>OuiText</OuiCode> it is often useful to apply floats.
            Almost always you&apos;ll want to pair the <OuiCode>float</OuiCode>{' '}
            prop usage, with a <OuiCode>margin</OuiCode> prop usage to give
            space around your image. Margins, when used in combo with floats,
            will adjust depending upon the position of the float.
          </p>
          <OuiCallOut title="Be careful with floats" color="warning">
            Floats should only be used on images within <strong>large</strong>{' '}
            bodies of text. Specifically, we only suggest using them with{' '}
            <OuiCode>OuiText</OuiCode> which comes automatically clears floats.
          </OuiCallOut>
        </Fragment>
      ),
      demo: <ImageFloat />,
      snippet: imageFloatSnippet,
    },
  ],
  playground: imageConfig,
};
