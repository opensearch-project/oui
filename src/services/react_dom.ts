/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
// eslint-disable-next-line import/no-unresolved
import { createRoot } from './react-dom-client';

/**
 * Version-agnostic render function that works with both React 16 and 18
 * In React 16, it uses our shim for react-dom/client (./react-dom-client.ts)
 * In React 18, we can alias this file to the actual react-dom/client
 */
export function render(element: React.ReactElement, container: Element) {
  const root = createRoot(container);
  root.render(element);
  return root;
}
