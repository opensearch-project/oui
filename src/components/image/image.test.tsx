/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import { requiredProps } from '../../test';
import { keys } from '../../services';

import { OuiImage } from './image';

describe('OuiImage', () => {
  test('is rendered', () => {
    const { container } = render(
      <OuiImage alt="alt" size="l" url="/cat.jpg" {...requiredProps} />
    );

    expect(container).toMatchSnapshot();
  });

  test('is rendered and allows full screen', () => {
    const { container } = render(
      <OuiImage
        alt="alt"
        size="l"
        url="/cat.jpg"
        allowFullScreen
        {...requiredProps}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('is rendered with src', () => {
    const { container } = render(
      <OuiImage alt="alt" float="left" src="/cat.jpg" />
    );

    expect(container).toMatchSnapshot();
  });

  test('is rendered with a float', () => {
    const { container } = render(
      <OuiImage alt="alt" float="left" url="/cat.jpg" />
    );

    expect(container).toMatchSnapshot();
  });

  test('is rendered with a margin', () => {
    const { container } = render(
      <OuiImage alt="alt" margin="l" url="/cat.jpg" />
    );

    expect(container).toMatchSnapshot();
  });

  test('is rendered with custom size', () => {
    const { container } = render(
      <OuiImage alt="alt" size={50} url="/cat.jpg" />
    );

    expect(container).toMatchSnapshot();
  });

  test('is rendered with a node as the caption', () => {
    const { container } = render(
      <OuiImage alt="alt" caption={<span>caption</span>} url="/cat.jpg" />
    );

    expect(container).toMatchSnapshot();
  });

  describe('Full screen behaviour', () => {
    beforeEach(() => {
      render(
        <OuiImage
          alt="alt"
          size="l"
          url="/cat.jpg"
          allowFullScreen
          {...requiredProps}
          data-test-subj="ouiImage"
        />
      );

      const activateButton = screen.getByTestId('activateFullScreenButton');
      fireEvent.click(activateButton);
    });

    test('full screen image is rendered', () => {
      const overlayMask = screen.getByTestId('fullScreenOverlayMask');
      expect(overlayMask).toBeInTheDocument();

      // Find the image within the full screen overlay
      const fullScreenImage = screen.getAllByTestId('ouiImage')[1]; // Get the second instance which is in the overlay
      expect(fullScreenImage).toBeInTheDocument();
    });

    test('close using close icon', () => {
      const deactivateFullScreenBtn = screen.getByTestId(
        'deactivateFullScreenButton'
      );
      expect(deactivateFullScreenBtn).toBeInTheDocument();

      act(() => {
        fireEvent.click(deactivateFullScreenBtn);
      });

      expect(
        screen.queryByTestId('fullScreenOverlayMask')
      ).not.toBeInTheDocument();
    });

    test('close using ESCAPE key', () => {
      const deactivateFullScreenBtn = screen.getByTestId(
        'deactivateFullScreenButton'
      );
      expect(deactivateFullScreenBtn).toBeInTheDocument();

      act(() => {
        fireEvent.keyDown(deactivateFullScreenBtn, {
          key: keys.ESCAPE,
          bubbles: true,
        });
      });

      expect(
        screen.queryByTestId('fullScreenOverlayMask')
      ).not.toBeInTheDocument();
    });

    test('close using overlay mask', () => {
      const overlayMask = screen.getByTestId('fullScreenOverlayMask');
      expect(overlayMask).toBeInTheDocument();

      act(() => {
        fireEvent.click(overlayMask);
      });

      expect(
        screen.queryByTestId('fullScreenOverlayMask')
      ).not.toBeInTheDocument();
    });
  });
});
