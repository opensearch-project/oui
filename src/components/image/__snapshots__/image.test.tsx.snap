// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiImage is rendered 1`] = `
<div>
  <figure
    class="ouiImage testClass1 testClass2 ouiImage--large"
  >
    <img
      alt="alt"
      aria-label="aria-label"
      class="ouiImage__img"
      data-test-subj="test subject string"
      src="/cat.jpg"
    />
  </figure>
</div>
`;

exports[`OuiImage is rendered and allows full screen 1`] = `
<div>
  <figure
    class="ouiImage ouiImage--allowFullScreen testClass1 testClass2 ouiImage--large"
  >
    <button
      aria-label="Open full screen alt image"
      class="ouiImage__button ouiImage__button--fullWidth"
      data-test-subj="activateFullScreenButton"
      type="button"
    >
      <img
        alt="alt"
        aria-label="aria-label"
        class="ouiImage__img"
        data-test-subj="test subject string"
        src="/cat.jpg"
      />
      <span
        class="ouiImage__icon"
        color="ghost"
        data-ouiicon-type="fullScreen"
      />
    </button>
  </figure>
</div>
`;

exports[`OuiImage is rendered with a float 1`] = `
<div>
  <figure
    class="ouiImage ouiImage--floatLeft ouiImage--original"
  >
    <img
      alt="alt"
      class="ouiImage__img"
      src="/cat.jpg"
    />
  </figure>
</div>
`;

exports[`OuiImage is rendered with a margin 1`] = `
<div>
  <figure
    class="ouiImage ouiImage--marginLarge ouiImage--original"
  >
    <img
      alt="alt"
      class="ouiImage__img"
      src="/cat.jpg"
    />
  </figure>
</div>
`;

exports[`OuiImage is rendered with a node as the caption 1`] = `
<div>
  <figure
    aria-label="caption"
    class="ouiImage ouiImage--original"
  >
    <img
      alt="alt"
      class="ouiImage__img"
      src="/cat.jpg"
    />
    <figcaption
      class="ouiImage__caption"
    >
      <span>
        caption
      </span>
    </figcaption>
  </figure>
</div>
`;

exports[`OuiImage is rendered with custom size 1`] = `
<div>
  <figure
    class="ouiImage"
  >
    <img
      alt="alt"
      class="ouiImage__img"
      src="/cat.jpg"
      style="max-width: 50px; max-height: 50px; width: auto;"
    />
  </figure>
</div>
`;

exports[`OuiImage is rendered with src 1`] = `
<div>
  <figure
    class="ouiImage ouiImage--floatLeft ouiImage--original"
  >
    <img
      alt="alt"
      class="ouiImage__img"
      src="/cat.jpg"
    />
  </figure>
</div>
`;
